elastalert/ruletypes.py

```diff
@@ -1286,6 +1285,11 @@ def __init__(self, *args):
         if not self.rules['pipeline_agg_type'] in self.allowed_pipeline_aggregations:
             raise EAException("pipeline_agg_type must be one of %s" % (str(self.allowed_pipeline_aggregations)))
 
+        assert total_seconds(self.rules['run_every']) == total_seconds(self.rules['bucket_interval_timedelta']), \
+                'bucket_interval must be exactly the same as run_every'
```

Comment:
Not totally sure this is necessary since all events within the `timeframe` are added into an [event window](https://github.com/BigDataBoutique/elastalert/blob/cd891e7e43d0263d8c77ea0b6bb0d53d41361a96/elastalert/ruletypes.py?plain=1#L308) and the value is then derived by averaging all events within the event window.

perhaps the following is more appropriate?

```
assert total_seconds(self.rules['run_every']) <= total_seconds(self.rules['bucket_interval_timedelta']), \
                'bucket_interval must be exactly the same as run_every'
```

Reply:
go for it